system_tests/block_validator_test.go

```diff
@@ -108,6 +112,16 @@ func testBlockValidatorSimple(t *testing.T, dasModeString string, workloadLoops
 		}
 	} else {
 		auth := l2info.GetDefaultTransactOpts("Owner", ctx)
+		// deploy a test contract
+		var err error
+		_, _, simple, err = mocksgen.DeploySimple(&auth, l2client)
+		Require(t, err, "could not deploy contract")
+
+		difficulty, err := simple.GetBlockDifficulty(&bind.CallOpts{})
```

Comment:
hey sorry, i am not clear on this part.
Can please provide an example transaction that should be made here? Also what do you mean when you say "verified correctly before and after update." ?

Reply:
Call does not change state and isn't included in a block, while transactions are.
For example: you can have a smart contract function call DIFFICULTY and store the result (you can also have another function return that value from storage - and use that other function to test that the transaction was executed correctly)
We need to execute that transaction twice - once before the upgrade, once after. This will also produce two blocks, accordingly, and we'll need to verify both of them (last part should not be a problem - the test should already verify all blocks created)

Reply:
thanks. Done.